src/hotspot/share/c1/c1_LinearScan.cpp

```diff
@@ -5752,6 +5752,7 @@ void LinearScanWalker::alloc_locked_reg(Interval* cur) {
 
 bool LinearScanWalker::no_allocation_possible(Interval* cur) {
 #ifdef X86
+#ifndef _LP64
```

Comment:
rbp is callee-saved, so the following logic doesn't work. That'll slow down the allocation.